File: PocketEdition/RoleRollsPocketEdition/_Domain/Itens/Templates/Models/ItemTemplateModel.cs
using RoleRollsPocketEdition._Application.Itens.Dtos;

namespace RoleRollsPocketEdition._Domain.Itens.Templates.Models;

public class ItemTemplateModel
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public Guid? PowerId { get; set; }
    public ItemType Type { get; set; }
    public Guid? CampaignId { get; set; }


    public static T FromTemplate<T>(ItemTemplate entity) where T : ItemTemplateModel, new()
    {
        return new T
        {
            Id = entity.Id,
            Name = entity.Name,
            PowerId = entity.PowerId,
            Type = entity.Type,
            CampaignId = entity.CampaignId,
        };
    }
}

File: frontend/role-rolls/src/app/models/ItemTemplateModel.ts
import { EquipableSlot } from 'app/models/itens/equipable-slot';

export interface ItemTemplateModel{
  id: string;
  campaignId: string;
  powerId: string;
  name: string;
  type: ItemType;
}
export interface EquipableTemplateModel extends ItemTemplateModel {
  slot: EquipableSlot;
}
export interface WeaponTemplateModel extends EquipableTemplateModel{
  category: WeaponCategory;
}

export interface ArmorTemplateModel extends EquipableTemplateModel{
  category: ArmorCategory;
}
export enum ItemType {
  Consumable = 0,
  Weapon = 1,
  Armor = 2,
}
export enum WeaponCategory
{
  Light = 0,
  Medium = 1,
  Heavy = 2,
}
export enum ArmorCategory
{
  Light = 0,
  Medium = 1,
  Heavy = 2,
}
